.github/workflows/neuron-ci.yml

```diff
@@ -0,0 +1,243 @@
+          - { matrix_eval : "CC=gcc-6 CXX=g++-6",   build_mode: "cmake",      cmake_option: "-DNRN_ENABLE_MPI=ON -DNRN_ENABLE_INTERVIEWS=ON -DNRN_ENABLE_CORENEURON=OFF" }
+          - { matrix_eval : "CC=gcc-9 CXX=g++-9",   build_mode: "cmake",      cmake_option: "-DNRN_ENABLE_PYTHON=OFF -DNRN_ENABLE_RX3D=OFF -DNRN_ENABLE_CORENEURON=ON" }
+          - { matrix_eval : "CC=gcc-8 CXX=g++-8",   build_mode: "cmake",      python_dynamic : ON }
+      fail-fast: false
+      
+    steps:
+      - name: Setup cmake
+        uses: jwlawson/actions-setup-cmake@v1.4
+        with:
+          cmake-version: ${{ env.DESIRED_CMAKE_VERSION }}
+
+      - name: Install homebrew packages
+        if: startsWith(matrix.os, 'macOS')
+        run: |
+          brew install coreutils doxygen mpich xz autoconf automake libtool && brew unlink mpich && brew install openmpi && brew cask install xquartz
+        shell: bash
+        
+      - name: Install apt packages
+        if: startsWith(matrix.os, 'ubuntu')
+        run: |
+          sudo apt-get install  build-essential doxygen autoconf automake libtool autotools-dev libopenmpi-dev libmpich-dev libx11-dev libxcomposite-dev mpich openmpi-bin patchelf python-pip python-tk python-numpy python3-numpy python3-pip g++-5 g++-6 g++-8 g++-9  
+        shell: bash
+
+      - name: Set up Python2
+        uses: actions/setup-python@v2
+        with:
+          python-version: '2.x'
+
+      - name: Install Python2 dependencies
+        if: ${{ matrix.config.use_python2 == 'ON' || matrix.config.python_dynamic == 'ON' }}
+        run: python2 -m pip install --upgrade pip wheel setuptools scikit-build pytest matplotlib bokeh ipython cython pytest mpi4py
+
+      - name: Set up Python3
+        uses: actions/setup-python@v2
+        with:
+          python-version: '3.8'
+
+      - name: Install Python3 dependencies
+        run: python3 -m pip install --upgrade pip wheel setuptools scikit-build pytest matplotlib bokeh ipython cython pytest mpi4py
+     
+      - uses: actions/checkout@v2
+        
+      - name: Install Python3 documentation dependencies
+        if: ${{ startsWith(matrix.os, 'ubuntu') && matrix.config.documentation == 'ON' }}
+        working-directory: ${{runner.workspace}}/nrn
+        run: python3 -m pip install --upgrade -r docs/docs_requirements.txt
+        
+      - name: Build and Test
+        id: build-test
+        shell: bash
+        working-directory: ${{runner.workspace}}/nrn
+        run:  |
+
+          # OS related
+          if [ "$RUNNER_OS" == "Linux" ]; then
+            eval "${{ matrix.config.matrix_eval }}";
+            export TRAVIS_OS_NAME="linux"
+            export SHELL="/bin/bash"
+          else
+            export CXX=${CXX:-g++};
+            export CC=${CC:-gcc};
+          fi
+          if [ "$RUNNER_OS" == "macOS" ]; then
+            export TRAVIS_OS_NAME="osx"
+            # TODO - this is a workaround that was implemented for Azure being reported as getting stuck.
```

Comment:
@pramodk we should tackle this in the near future

Reply:
@alexsavulescu : can we change here TRAVIS_OS_NAME and instead update shell script to use RUNNER_OS env name as well? (just this look bit confusing to set travis related env vars)

Reply:
I want to have a separate PR called `adjust scripts for CI synergy`, will rename everything there (there are a bunch of scripts to be updated)